Token.js

```diff
@@ -2778,20 +2782,55 @@ function determine_hidden_classname(tokenIds) {
 }
 
 function token_menu() {
-
-		$("#tokens").on("contextmenu", ".VTTToken", function(event) {
-			console.log("context_menu_flyout contextmenu event", event);
+	$("#tokens").on("contextmenu", ".VTTToken", function(event) {
 			event.preventDefault();
 			event.stopPropagation();
-			if($(event.currentTarget).children('.token-image').css('pointer-events') == 'none' && !window.DM)
+
+			let existingPopup = document.getElementById("tokenOptionsPopup");
+			if (existingPopup) {
+				remove_token_options_popup();
+			}
+
+			let target = $(event.currentTarget);
+			if(target.children('.token-image').css('pointer-events') == 'none' && !window.DM)
 				return;
-			if ($(event.currentTarget).hasClass("tokenselected") && window.CURRENTLY_SELECTED_TOKENS.length > 0) {
+			if (target.hasClass("tokenselected") && window.CURRENTLY_SELECTED_TOKENS.length > 0) {
 				token_context_menu_expanded(window.CURRENTLY_SELECTED_TOKENS, event);
 			} else {
 				token_context_menu_expanded([$(event.currentTarget).attr("data-id")], event);
 			}
-		});
-		return;
+
+			window.addEventListener('click', remove_token_options_popup);
+	});
+}
+
+function remove_token_options_popup(event = null) {
+	if (!event) {
+		$("#tokenOptionsPopup").remove();
+		$('.context-menu-list').trigger('contextmenu:hide')
+		//tokenOptionsClickCloseDiv.remove();
+		$("#tokenOptionsContainer .sp-container").spectrum("destroy");
+		$("#tokenOptionsContainer .sp-container").remove();
+		$(`.context-menu-flyout`).remove();
+		window.removeEventListener('click', remove_token_options_popup);
+	}
+
+	let target = event.target;
+	let clickedMenuInput = () => target.matches(".elevMenuInput");
+	let clickedMenuFlyoutButton = () => target.matches(".token-image-modal-footer-title");
+	let clickedConditions = () => !!target.parentNode.closest("#conditions-flyout");
+	let didNotClickAdjustmentsAndTokenImage = () => !!target.parentNode.closest("#adjustments-flyout") && !target.matches("#changeTokenImage");
+	let didNotClickLight = () => !!target.parentNode.closest("#light-flyout");
+	let didNotClickOptions = () => !!target.parentNode.closest("#options-flyout");
+	if (!(clickedMenuInput() || clickedMenuFlyoutButton() || clickedConditions() || didNotClickAdjustmentsAndTokenImage() || didNotClickLight() || didNotClickOptions())) {
```

Comment:
The hp/max hp and ac inputs closes it still. Aura flyout stuff closes it. 

I personally feel the only things it needs to close for are, the group roll, notes and delete.

Maybe a few other buttons I could see being one time clicks (like token image since we're interacting with the side panel after). So basically anything where we're interacting with a new window/area outside the menu immediately after clicking it or removing the token.

I feel it should be consistent where possible and some of the buttons I use in tandem.

I may want to unhide a bunch of monsters while adding them to the combat tracker during an ambush for example. Or add a single monster to the tracker and open it's sheet. 

I think the trade off of a click off to close feels better than having to reopen and click for each of those while testing this personally. It could just be my preference though and I'm not sure how it feels using a trackpad or something other than a mouse.

Reply:
I think I got everything you asked for. The hiding/unhide feels like we should push them to use the multiselect vs leaving it open and letting them click over right?

Reply:
What do you mean by push them to multiselect here? Sorry Im on my phone so I havent been able to test it yet.

The below examples was what I meant when talking about hiding stuff. You may have already changed it this way and your talking about something else.

If i have a group of goblins come out from behind trees abushing the party for example. Id want to unhide their tokens and add them to combat.

If i have a purple worm explode out of the ground I want to be able to do all three, unhide, add to combat and open monster stat block. 

I feel like we really dont know when a dm/player might be done with the menu. So in general I feel it should only close when we know they arent going to want to use another button/option.

Reply:
Ah I see what you are saying with the combat and open monster stat block piece. Ill make those changes. Thanks for laying that out.